### PR TITLE
React data tables

### PR DESCRIPTION
**Pull request by paweltin**

### CODE REVIEW

File: saleor/static/js_src/dashboard.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        this.setState({})
+    };
+
+    render() {
+        return <DataGrid
+            idProperty={this.props.columns[0]}
+            dataSource={this.state.data}
+            columns={this.props.columns}
+            sortInfo={SORT_INFO}
+            onSortChange={this.handleSortChange.bind(this)}
```
Comment (patrys) [inline saleor/static/js_src/dashboard.jsx]: Could be `{::this.handleSortChange}` or we could import the [`autobind-decorator`](https://github.com/andreypopp/autobind-decorator) decorator package.

File: Gruntfile.js [hunk 2]
```diff
@@ -3,7 +3,8 @@ module.exports = function(grunt) {
     babel: {
       dist: {
         files: {
-          "saleor/static/js/storefront_fromjsx.js": "saleor/static/js_src/storefront.jsx"
+          "saleor/static/js/storefront_fromjsx.js": "saleor/static/js_src/storefront.jsx",
```
Comment (patrys) [inline Gruntfile.js]: Is there a reason to have `_fromjsx` in file names?

File: saleor/order/forms.py [hunk 3]
```diff
@@ -3,12 +3,14 @@
 from django.utils.translation import ugettext_lazy as _
 
 from .models import Payment
+from .widgets import ButtonSelect
```
Comment (patrys) [inline saleor/order/forms.py]: Where is this?

File: saleor/static/js_src/dashboard.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        SORT_INFO = sortInfo;
+
+        this.setState({data: sort(this.props.data)});
+    };
+
+    handleColumnOrderChange(index, dropIndex) {
+        var col = columns[index];
+        columns.splice(index, 1); //delete from index, 1 item
+        columns.splice(dropIndex, 0, col);
+        this.setState({})
```
Comment (patrys) [inline saleor/static/js_src/dashboard.jsx]: Missing semicolon.

Comment (patrys) [inline saleor/static/js_src/dashboard.jsx]: Also, what is the point of setting an empty state? There is a `forceUpdate()` method if you really need it but `columns` definitely should be part of the state. Components should never depend on data outside of their state and props (here `columns` seems to be global :skull:).

File: saleor/static/js_src/dashboard.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    };
+
+    render() {
+        return <DataGrid
```
Comment (patrys) [inline saleor/static/js_src/dashboard.jsx]: Here you depend on something that was never imported.

File: saleor/order/forms.py (hunk 3 above)
Comment (paweltin) [inline saleor/order/forms.py]: Forgot to commit files. Fixed.

File: bower.json [hunk 6]
```diff
@@ -19,6 +19,7 @@
     "modernizr": "~2.8.3",
     "react": "~0.13.3",
     "select2": "~4.0.0",
-    "Sortable": "~1.2.0"
+    "Sortable": "~1.2.0",
+    "mdi": "~1.2.65"
```
Comment (patrys) [inline bower.json]: Please sort.

File: package.json [hunk 7]
```diff
@@ -29,6 +29,10 @@
     "grunt-contrib-watch": "~0.6.1",
     "grunt-postcss": "~0.4.0",
     "grunt-sass": "~1.0.0",
-    "load-grunt-tasks": "~3.2.0"
+    "load-grunt-tasks": "~3.2.0",
+    "react": "~0.13.3",
+    "react-datagrid": "~1.2.13",
+    "sorty": "~1.2.0",
+    "grunt-browserify": "~4.0.0"
```
Comment (patrys) [inline package.json]: Please sort.

File: saleor/dashboard/order/api.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
+from rest_framework import serializers
+from rest_framework.fields import SerializerMethodField
+from rest_framework.filters import DjangoFilterBackend
+from rest_framework.viewsets import ModelViewSet
+
+from ...order.models import Order
+
+
+class OrderSerializer(serializers.ModelSerializer):
+
```
Comment (patrys) [inline saleor/dashboard/order/api.py]: We can drop this empty line.

File: saleor/dashboard/order/api.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    def get_total(self, obj):
+        total = obj.get_total()
+        price = {name: str(value) for name, value in total._asdict().items()}
+        del price['history']
+        return price
+
+
+class OrderViewSet(ModelViewSet):
+
```
Comment (patrys) [inline saleor/dashboard/order/api.py]: Same for this one.

File: saleor/dashboard/templates/dashboard/includes/_orders_table.html [hunk 10]
```diff
@@ -3,15 +3,15 @@
 {% load status %}
 
 <div class="data-table-container">
-    <table class="bordered hoverable responsive data-table">
+    <table class="bordered hoverable responsive data-table{% if not not_sortable %} data-table-sortable{% endif %}">
```
Comment (patrys) [inline saleor/dashboard/templates/dashboard/includes/_orders_table.html]: `not_sortable`? `not_not_sortable`?

File: saleor/static/js_src/dashboard.jsx [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+        });
+    };
+
+    render() {
+        return <DataGrid
+            idProperty={this.props.columns[0]+""}
+            dataSource={this.state.data}
+            columns={this.props.columns}
+            sortInfo={this.state.sortInfo}
+            onSortChange={this.handleSortChange.bind(this)}
```
Comment (patrys) [inline saleor/static/js_src/dashboard.jsx]: Babel supports binding through `::` operator:

```
onSortChange={::this.handleSortChange}
```

(This is the same as `this::this.handleSortChange`.)

**Comment by paweltin** [comment]
@patrys I think it's ready for review. Async filtering is not working as backend for it has yet to be done.

**Pull request closed without merging**